streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -1295,6 +1297,78 @@ object ZSink extends ZSinkPlatformSpecific {
         }
     }
 
+  /**
+   * Splits strings on newlines. Handles both `\r\n` and `\n`.
+   */
+  final val splitLines: ZSink[Any, Nothing, String, String, Chunk[String]] =
```

Comment:
Not to nitpick here, but I see this as a deviation from the previous sinks, which have all been defined with `def`. AFAIK `val` creates objects eagerly, and they stay in memory for the duration of the program which references this code. Is that something desirable in this case? I may be completely off base here, I just want to hear the reasoning. Same for every other `val`, including the ones in `Sink`.